product-mini/platforms/posix/main.c

```diff
@@ -696,10 +696,10 @@ main(int argc, char *argv[])
     /* destroy runtime environment */
     wasm_runtime_destroy();
 
-#if WASM_ENABLE_SPEC_TEST != 0
-    (void)ret;
-    return 0;
-#else
+    // #if WASM_ENABLE_SPEC_TEST != 0
+    //     (void)ret;
+    //     return 0;
+    // #else
     return ret;
-#endif
+    // #endif
```

Comment:
how is this related to the rest of the PR?

Reply:
In [wasi-testsuite](https://github.com/WebAssembly/wasi-testsuite) there are some tests that check the return code and, because of the hardcoded `return 0`, would fail.
It has been fixed in the `main` branch now #1377 #1839.

To merge this PR, we need https://github.com/WebAssembly/wasi-testsuite/pull/40 to be merged and `main` to be merged into [dev/wasi_threads](https://github.com/bytecodealliance/wasm-micro-runtime/tree/dev/wasi_threads).

Reply:
@wenyongh would you be able to help with the second point and merge `main` to `dev/wasi_threads` and unblock this PR?

Reply:
@eloparco @loganek I have merged `main` into `dev/wasi_threads`, since there are code conflicts, I applied the merging manually, please check whether it works for you:
 https://github.com/bytecodealliance/wasm-micro-runtime/commit/3403f0a7f4d838f260b55b327da8be00a8644229

Reply:
@wenyongh thanks a lot, do you remember which files required manual conflict resolution?

Reply:
@loganek Not the files related to wasi-threads, but some files modified in main branch. I used mode `rebase and merge` instead of mode `merge` to merge `main` into `dev/wasi_threads` last time, it didn't produce "a new generated commit" and seems that `dev/wasi_threads` didn't remember the sync-up status from `main`, so when the same line of the same file was modified in main branch by new commit, conflict occurs. Similar conflicts occurred when I tried merging main into `dev/fast_jit_simd` and `gitbook`, I think I should use `merge` mode to merge main into other branches next time.

Reply:
ok, thanks. Looks good to me. I think once we're ready with the dev branch, I'll rebase it completely on top of main and open a final PR to rebase-merge it, to keep the git tree clean.